deployment/keystone/deploy_test.go

```diff
@@ -100,7 +103,7 @@ func TestDeploy(t *testing.T) {
 	maps.Copy(allNodes, wfNodes)
 	maps.Copy(allNodes, cwNodes)
 	maps.Copy(allNodes, assetNodes)
-	env := memory.NewMemoryEnvironmentFromChainsNodes(t, lggr, allChains, allNodes)
+	env := memory.NewMemoryEnvironmentFromChainsNodes(func() context.Context { return ctx }, lggr, allChains, allNodes)
```

Comment:
When these constructors are called normally, is there always an appropriate context on hand to pass in already? If not (or even if so?), this signature could be simplified by making a new context internally and returning (or providing a method to invoke) `cancel()`.
```suggestion
	env, cancel := memory.NewMemoryEnvironmentFromChainsNodes(lggr, allChains, allNodes)
	t.Cleanup(cancel)
```
This is also compatible with cases where there is a context already:
```suggestion
	env := memory.NewMemoryEnvironmentFromChainsNodes(lggr, allChains, allNodes)
	context.AfterFunc(ctx, env.Cancel)
```

Reply:
There is already a context in the other case - I plan to normalize them.  I am trying to keep things somewhat stable. I have a lot of change to stack on this, and I'm trying to not change too much of the signatures in one PR. I want to do a pass through all of this. They were pulling a context off of testing.T internally, and this was intended to make them both work the same way, pending future fixes. I'm not trying to fix all the things in this PR. 

This is a good pattern to keep in mind, however, and I'll look at it as an option for a followup.

Reply:
Also, that doesn't exactly simplify the signature, it just moves a parameter to a return type, so it's the same number of parameters (if you count return values as "out" params)

Reply:
It does simplify having to construct a lambda, but I have a cleanup there planned as well.

Reply:
Please consider the method variant.

Reply:
Per our discussion, I'll definitely look at the method variant, but for now, plumbing through the lambda is enough to unblock other work, and I'll make a point of revisiting this for a more mature signal handling once we're out of the crunch.